framework/src/org/apache/cordova/CordovaWebViewClient.java

```diff
@@ -89,6 +97,15 @@ public void setWebView(CordovaWebView view) {
         this.appView = view;
         helper = new CordovaUriHelper(cordova, view);
     }
+    
+    /**
+     * Specifies the HTTP auth handler for handling auth challenge requests.
+     *
+     * @param handler HTTP auth handler.
+     */
+    public void setHttpAuthRequestHandler(HttpAuthRequestHandler handler) {
```

Comment:
Why use a setter here instead of following the plugin pattern of allowing all plugins to handle the callback?